app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [Person(person["name"], person["age"]) for person in people]
+
+    for person_dict in people:
+        person = Person.people[person_dict["name"]]
+
+        spouse_key = "wife" if "wife" in person_dict else "husband"
+        spouse_name = person_dict.get(spouse_key)
```

Comment:
The logic for determining the spouse key assumes that a person can only have either a 'wife' or a 'husband'. If both keys are present in the dictionary, only the 'wife' key will be used. Ensure that this logic aligns with the intended data structure and requirements.